pipreqs/pipreqs.py

```diff
@@ -102,13 +122,26 @@ def get_all_imports(path, encoding=None, extra_ignore_dirs=None, follow_links=Tr
         dirs[:] = [d for d in dirs if d not in ignore_dirs]
 
         candidates.append(os.path.basename(root))
-        files = [fn for fn in files if os.path.splitext(fn)[1] == ".py"]
+        if PythonExporter and not ignore_notebooks:
+            files = [fn for fn in files if filter_ext(fn, [".py", ".ipynb"])]
+        else:
+            files = [fn for fn in files if filter_ext(fn, [".py"])]
+
+        candidates = list(
+            map(
+                lambda fn: os.path.splitext(fn)[0],
+                filter(lambda fn: filter_ext(fn, [".py"]), files),
+            )
+        )
 
-        candidates += [os.path.splitext(fn)[0] for fn in files]
         for file_name in files:
             file_name = os.path.join(root, file_name)
-            with open(file_name, "r", encoding=encoding) as f:
-                contents = f.read()
+            contents = ""
+            if filter_ext(file_name, [".py"]):
+                with open(file_name, "r", encoding=encoding) as f:
+                    contents = f.read()
+            elif filter_ext(file_name, [".ipynb"]) and PythonExporter and not ignore_notebooks:
```

Comment:
not the first time I see this condition `PythonExporter and not ignore_notebooks`.

maybe we could create a function with this condition so that its meaning is clearer. Example:

```
def notebooks_are_enabled():
   return PythonExporter and not ignore_notebooks
```

Reply:
Done